File: src/main/java/xyz/theprogramsrc/superauth_v3/spigot/objects/AuthMethod.java
package xyz.theprogramsrc.superauth_v3.spigot.objects;

public enum AuthMethod {
    DIALOG,
    GUI,
    COMMANDS

    ;

    public static AuthMethod of(String name){
        if(name == null) return DIALOG;
        if(name.equals(" ") || name.equals("") || name.isEmpty()) return DIALOG;
        try{
            return valueOf(name);
        }catch (Exception ex){
            return DIALOG;
        }
    }
}

File: src/main/kotlin/xyz/theprogramsrc/superauth/global/AuthMethod.kt
package xyz.theprogramsrc.superauth.global

/**
 * Representation of an authentication method
 */
enum class AuthMethod {
    /**
     * Authentication with PIN GUI
     */
    GUI,

    /**
     * Authentication with Chat Password
     */
    DIALOG,

    /**
     * Authentication with Command Password
     */
    COMMAND,
}